app/extensions/safe/ffi/authenticator.js

```diff
@@ -86,6 +86,18 @@ class Authenticator extends SafeLib
         this[_registeredClientHandle] = handle;
     }
 
+    get reAuthoriseState()
+    {
+      return this[_reAuthoriseState];
+    }
+
+    set reAuthoriseState( state )
+    {
+      logger.info('-----------------------------------------------');
+      logger.info('old reAuthoriseState: ', this.reAuthoriseState);
+      logger.info('new reAuthoriseState: ', state);
```

Comment:
I'm able to observe a successful setting of `reAuthoriseState` for the `Authenticator` instance when clicking on lock icon. Subsequent clicks will correctly show the previous state.
However, icon when is unlocked, `reAuthoriseState`  has value of 1, and I authorise an app, the notification still appears because `this[_reAuthoriseState]` has an unexpected value of 0.